src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -2093,20 +2089,37 @@ internal override BaseClientSessionTransportManager CreateClientSessionTransport
                         StringUtil.Format(RemotingErrorIdStrings.KeyFileNotFound, this.KeyFilePath));
                 }
 
-                arguments = (string.IsNullOrEmpty(domainName)) ?
-                    string.Format(CultureInfo.InvariantCulture, @"-i ""{0}"" {1}@{2} -p {3} -s {4}", this.KeyFilePath, userName, hostname, this.Port, this.Subsystem) :
-                    string.Format(CultureInfo.InvariantCulture, @"-i ""{0}"" -l {1}@{2} {3} -p {4} -s {5}", this.KeyFilePath, userName, domainName, hostname, this.Port, this.Subsystem);
+                arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-i ""{0}""", this.KeyFilePath));
             }
-            else
+
+            // pass "-l login_name" commmand line argument to ssh if UserName is set
+            if (!string.IsNullOrEmpty(this.UserName))
             {
-                arguments = (string.IsNullOrEmpty(domainName)) ?
-                    string.Format(CultureInfo.InvariantCulture, @"{0}@{1} -p {2} -s {3}", userName, hostname, this.Port, this.Subsystem) :
-                    string.Format(CultureInfo.InvariantCulture, @"-l {0}@{1} {2} -p {3} -s {4}", userName, domainName, hostname, this.Port, this.Subsystem);
+                var parts = this.UserName.Split(Utils.Separators.Backslash);
+                if (parts.Length == 2)
+                {
+                    // ? pass DOMAIN\username as username@DOMAIN (still needed?)
+                    arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-l {0}@{1}", parts[1], parts[0]));
+                }
+                else
+                {
+                    arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-l {0}", this.UserName));
+                }
             }
 
+            // pass "-p port" command line argument to ssh if Port is set
+            if (this.Port != ReservedPort)
+            {
+                arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-p {0}", this.Port));
+            }
+
+            // pass "-s destination command" command line arguments to ssh where command is the subsystem to invoke on the destination
+            // ? note that ssh expects IPv6 addresses to not be enclosed in square brackets so trim them (still needed?)
+            arguments.Add(string.Format(CultureInfo.InvariantCulture, @"-s {0} {1}", this.ComputerName.TrimStart('[').TrimEnd(']'), this.Subsystem));
+
             System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo(
                 filePath,
-                arguments);
+                string.Join(" ", arguments));
```

Comment:
Can we use the ArgumentList member of ProcessStartInfo instead of Arguments?  This would prevent any potential string injection.

Reply:
Good idea.